microprofile/config-smallrye/src/main/java/org/wildfly/extension/microprofile/config/smallrye/MicroProfileSubsystemDefinition.java

```diff
@@ -52,6 +54,7 @@ protected MicroProfileSubsystemDefinition(Iterable<ConfigSourceProvider> provide
                 .setAddHandler(new MicroProfileConfigSubsystemAdd(providers, sources))
                 .setRemoveHandler(ReloadRequiredRemoveStepHandler.INSTANCE)
                 .setCapabilities(CONFIG_CAPABILITY)
+                .setAdditionalPackages(RuntimePackageDependency.optional("org.jboss.resteasy.microprofile.config"))
```

Comment:
I don't think this will result in it being provisioned if layers are used. RuntimePackageDependency.optional should.